chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenario.java

```diff
@@ -45,15 +45,19 @@ public QueryPoolScenario(Invoker<Q, R, E> invoker, Iterator<Pair<Q, E>> chunks,
         this.chunks = chunks;
         this.systemClock = systemClock;
     }
-
+    
     @Override
-    public void doTransaction() {
+    public boolean doTransaction() {
         Pair<Q, E> chunk = chunks.next();
+        if (chunk == null) {
+            return false;
+        }
+        
         Q query = chunk.getFirst();
         E endpoint = chunk.getSecond();
-
+        
         invoker().invoke(query, endpoint);
-
+        return true;
```

Comment:
Why? As I know invoker().invoke(query, endpoint) may produce an exception why then the result is true?